src/app/controllers/notifications_controller.rb

```diff
@@ -0,0 +1,31 @@
+class NotificationsController < ApplicationController
+  LIMIT = 30
+
+  def index
+    offset = params.fetch(:offset, 0).to_i
+    @notifications = current_user.notifications.limit(LIMIT).offset(offset).order(created_at: :desc).unread
+    @next = offset + LIMIT if @notifications.size == LIMIT
+  end
+
+  def show
+    notification = current_user.notifications.find(params[:id]) || raise(ActiveRecord::RecordNotFound)
+    notification.update!(read_at: DateTime.now) unless notification.read_at
+
+    redirect_to_notification notification
+  end
+
+  private
+
+  def redirect_to_notification(notification)
+    notifiable = notification.notifiable
+    case notification.notifiable_type
+    when 'Like'
+      function = current_user.functions.find_by(id: notifiable.likeable_id) || raise(ActiveRecord::RecordNotFound)
+      redirect_to user_function_path(id: function, user_id: function.user)
+    when 'Comment'
+      redirect_to action: :index
+    else
+      redirect_to action: :index
+    end
```

Comment:
This function doesn't look in the right place

Reply:
Where do you suggest it should be moved to?